### PR TITLE
feat: add single selection bundle dropdown

### LINKED ISSUE #10115: [Łukasz] Create component for single-select drop-down bundle UI type
BI with Mockups: https://cxjira.sap.com/browse/TIGER-7006

There are a number of DOM and CCS requirements that could be found [here](https://wiki.wdf.sap.corp/wiki/display/CPQ/Clean+up+DOM).

### PR DESCRIPTION
**Pull request by luklus**
close #10115

### CODE REVIEW

File: feature-libs/product/configurators/rulebased/core/model/configurator.model.ts [hunk 1]
```diff
@@ -145,24 +145,24 @@ export namespace Configurator {
   }
 
   export enum UiType {
```
Comment (luklus) [inline feature-libs/product/configurators/rulebased/core/model/configurator.model.ts:147]: sort for better readability

File: feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<div
+  id="{{ createAttributeIdForConfigurator(attribute) }}"
+  class="form-group"
+  *ngIf="attribute?.values?.length !== 0"
```
Comment (kacperknapik) [inline feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.html]:
```suggestion
  *ngIf="attribute?.values?.length"
```

`0` is a falsy value, so we don't need to check for `!== 0`

File: feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+import { Configurator } from '../../../../core/model/configurator.model';
+import { ConfiguratorAttributeBaseComponent } from '../base/configurator-attribute-base.component';
+import { ConfiguratorAttributeSingleSelectionBundleDropdownComponent } from './configurator-attribute-single-selection-bundle-dropdown.component';
+import { ConfiguratorAttributeProductCardComponent } from '../../product-card/configurator-attribute-product-card.component';
+import { ConfiguratorShowMoreComponent } from '../../../show-more/configurator-show-more.component';
+import { I18nTestingModule } from '@spartacus/core';
+import { RouterTestingModule } from '@angular/router/testing';
+import { UrlTestingModule } from 'projects/core/src/routing/configurable-routes/url-translation/testing/url-testing.module';
+
+describe('ConfiguratorAttributeSingleSelectionBundleDropdownComponent', () => {
```
Comment (kacperknapik) [inline feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.spec.ts:20]: these names are really long. we should consider changing the name convention here

File: feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
+import { Configurator } from '../../../../core/model/configurator.model';
```
Comment (kacperknapik) [inline feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.ts:10]: is this import correct? I mean, isn't it importing from outside the lib? please check

**Review by kacperknapik — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.ts (hunk 4 above)
Comment (luklus) [reply feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.ts:10]: there are imports to product-configurator/rulebased/core/model files so are in same lib

File: feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.spec.ts (hunk 3 above)
Comment (luklus) [reply feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.spec.ts:20]: it is true, name convention is similar in all form types components, so i think we should rename it in other refactor task

**Review by Larisa-Staroverova — APPROVED** [review]

**Review by kacperknapik — APPROVED** [review]

**Pull request merged**